central/vulnmgmt/vulnerabilityrequest/datastore/datastore_impl.go

```diff
@@ -125,6 +129,21 @@ func (ds *datastoreImpl) UpdateRequestStatus(ctx context.Context, id, message st
 		return nil, errors.Errorf("vulnerability request %s not found", id)
 	}
 
+	if status == storage.RequestStatus_APPROVED && utils.V2GlobalScope(req.GetScope()) {
+		// Approvers of exceptions affecting all images should have access to all namespaces
+		allCount, err := ds.namespaceDatastore.Count(allAccessCtx, search.EmptyQuery())
+		if err != nil {
+			return nil, err
+		}
+		scopedCount, err := ds.namespaceDatastore.Count(ctx, search.EmptyQuery())
+		if err != nil {
+			return nil, err
+		}
+		if allCount != scopedCount {
+			return nil, errors.Wrap(errox.NotAuthorized, "Access scope of global exception request approvers should include all namespaces")
+		}
```

Comment:
Can this logic be extracted in a dedicated function ? I think that would improve the readability of the function flow.

Reply:
I think the logic here should focus on the namespaces the approvers are allowed to reach with read access to the `Image` resource (which controls image CVEs).

To achieve this, I think a SAC helper for the `Image` resource is needed.
If the access scope tree when reading `Image` related objects has full access (tree root included), then no need to check more closely the namespaces.
Otherwise, I think all namespaces should be pulled from the store, and access checked for each of them (drill-down in the effective access scope tree).
This kind of logic would better be implemented in a SAC helper object or function (with possibly the exception of retrieving all namespaces from the -cached- store). I'll try to come up with such an object shortly.

Reply:
With #14540 , the function could look like:
```
func ensureApproverCanSeeAllImages(ctx context.Context) error {
    imageSAC := sac.ForResource(resources.Image)
    hasReadAll, err := imageSAC.HasGlobalRead(ctx)
    if err != nil { return err }
    if hasReadAll { return nil }
    allNamespaces, err := ds.namespaceDatastore.GetAllNamespaces(allAccessCtx)
    if err != nil { return err }
    filteredNamespaces, err := imageSAC.FilterAccessibleNamespaces(ctx, storage.Access_READ_ACCESS, allNamespaces)
    if err != nil { return err }
    if len(allNamespaces) == len(filteredNamespaces) { return nil }
    return errors.Wrap(errox.NotAuthorized, "Access scope of global exception request approvers should include all namespaces")
}
```

Reply:
Thanks for creating #14540 . It is a good suggestion. I will wait for the PR to be merged and apply the suggested changes to this one.